auth_proxy/services/oauth.py

```diff
@@ -168,24 +185,15 @@ def cb_tokensetter(self, token, request, *args, **kwargs):
         """
         assert request.user is not None
 
-        try:
-            today = arrow.now().datetime
-            old = self.db.session.query(Token).\
-                filter_by(client_id=request.client.client_id).\
-                filter(Token.approval_expires >= today).\
-                one()
-            old.update_token(**token)
-        except NoResultFound:
-            old = Token(
-                token_type=token['token_type'],
-                _scopes=token['scope'],
-                client_id=request.client.client_id,
-                user=request.user,
-                approval_expires=arrow.now().shift(years=1).datetime,
-            )
-            old.update_token(**token)
-            self.db.session.add(old)
+        today = arrow.now().datetime
+        old = self.db.session.query(Token).\
```

Comment:
I'm just getting reacquainted with the flask-oauth structure; can you help me understand why the `NoResultFound` exception that you preciously caught isn't relevant anymore?